wire/Makefile

```diff
@@ -26,27 +26,62 @@ ALL_OBJS += $(WIRE_OBJS) $(WIRE_ONION_OBJS)
 ALL_GEN_HEADERS += $(WIRE_GEN_HEADERS)
 
 # They may not have the bolts.
-BOLT_EXTRACT=$(LOCAL_BOLTDIR)/tools/extract-formats.py
+EXTRACT_PATH=/tools/extract-formats.py
+BOLT_EXTRACT=$(LOCAL_BOLTDIR)$(EXTRACT_PATH)
 
 ifeq ($(DEVELOPER),1)
 wire/gen_peer_wire_csv: bolt-precheck
 	@set -e; if [ -d $(LOCAL_BOLTDIR) ]; then for f in $(LOCAL_BOLTDIR)/0[127]*.md $(BOLT_EXTRACT); do if [ $$f -nt $@ -o ! -f $@ ]; then $(BOLT_EXTRACT) --message-fields --message-types $(LOCAL_BOLTDIR)/0[127]*.md > $@; break; fi; done; fi
 
 wire/gen_onion_wire_csv: bolt-precheck
 	@set -e; if [ -f $(BOLT_EXTRACT) ]; then for f in $(LOCAL_BOLTDIR)/04*.md $(BOLT_EXTRACT); do if [ $$f -nt $@ -o ! -f $@ ]; then echo '#include <wire/onion_defs.h>' > $@ && $(BOLT_EXTRACT) --message-fields --message-types $(LOCAL_BOLTDIR)/04*.md >> $@; break; fi; done; fi
+
+ifdef EXP_BOLTDIR
+EXP_BOLT_EXTRACT=$(EXP_BOLTDIR)$(EXTRACT_PATH)
+
+wire/gen_peer_experimental_csv: wire/gen_peer_wire_csv
+	@if [ -d $(EXP_BOLTDIR) ]; then for f in $(EXP_BOLTDIR)/0[127]*.md $(EXP_BOLT_EXTRACT); do if [ $$f -nt $@ -o ! -f $@ ]; then $(EXP_BOLT_EXTRACT) --message-fields --message-types $(EXP_BOLTDIR)/0[127]*.md | diff -u wire/gen_peer_wire_csv - >>$@; break; fi; done; fi;
+
+wire/gen_onion_experimental_csv: wire/gen_onion_wire_csv
+	@if [ -d $(EXP_BOLTDIR) ]; then for f in $(EXP_BOLTDIR)/04*.md $(EXP_BOLT_EXTRACT); do if [ $$f -nt $@ -o ! -f $@ ]; then { echo '#include <wire/onion_defs.h>'; $(EXP_BOLT_EXTRACT) --message-fields --message-types $(EXP_BOLTDIR)/04*.md; } | diff -u $< - >>$@; break; fi; done; fi;
+
+spec-exp-peer: wire/gen_peer_experimental_csv
+spec-exp-onion: wire/gen_onion_experimental_csv
```

Comment:
these don't really work the way i want them to (regenerate the dependency file) but i didn't have much luck finding an alternative way to have Make re-run a 'named file target'. suggestions on how to fix this would be v welcome.

Reply:
ok updated, thanks for the suggestion @rustyrussell